src/platforms/rp2040/src/lib/networkdriver.c

```diff
@@ -419,6 +480,50 @@ static term start_ap(term ap_config, GlobalContext *global)
         }
     }
 
+    char *ap_hostname = driver_data->ap_hostname;
+    if (term_is_invalid_term(hostname_term)) {
+        ap_hostname = malloc(DEFAULT_HOSTNAME_SIZE);
+        if (IS_NULL_PTR(ap_hostname)) {
+            free(ssid);
+            free(psk);
+            return OUT_OF_MEMORY_ATOM;
+        } else {
+            term error = get_default_device_name(ap_hostname, global);
+            if (error != OK_ATOM) {
+                free(ssid);
+                free(psk);
+                free(ap_hostname);
+                return error;
+            }
+        }
+    } else {
+        int ok = 0;
+        char *buf = malloc(MAX_HOSTNAME_SIZE);
+        if (IS_NULL_PTR(buf)) {
+            free(ssid);
+            free(psk);
+            return OUT_OF_MEMORY_ATOM;
+        } else {
+            buf = interop_term_to_string(hostname_term, &ok);
+        }
+        if (!ok || IS_NULL_PTR(buf)) {
+            free(ssid);
+            free(psk);
+            return BADARG_ATOM;
+        }
+        ap_hostname = malloc(strlen(buf) + 1);
```

Comment:
I am very confused about when this is freed.

Reply:
This is freed when the driver is terminated.
```
void network_driver_destroy(GlobalContext *global)
{
    if (driver_data) {
        if (driver_data->hostname) {
            free(driver_data->hostname);
...
```